From: Ralf Baechle <ralf@linux-mips.org>
Date: Wed, 3 Dec 2008 17:03:27 +0000 (+0000)
Subject: MIPS: 64-bit: vmsplice needs to use the compat wrapper for o32 and N32.
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=85940fa4e1f9d731287d8836bc8f4948447b3fb5

MIPS: 64-bit: vmsplice needs to use the compat wrapper for o32 and N32.

Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit 1d12d2db84099dfd180cc314a3d8543098b53b4e)
---

diff --git a/arch/mips/kernel/scall64-n32.S b/arch/mips/kernel/scall64-n32.S
index 12940ec..72e1e9a 100644
--- a/arch/mips/kernel/scall64-n32.S
+++ b/arch/mips/kernel/scall64-n32.S
@@ -390,7 +390,7 @@ EXPORT(sysn32_call_table)
 	PTR	sys_splice
 	PTR	sys_sync_file_range
 	PTR	sys_tee
-	PTR	sys_vmsplice			/* 6270 */
+	PTR	compat_sys_vmsplice		/* 6270 */
 	PTR	sys_move_pages
 	PTR	compat_sys_set_robust_list
 	PTR	compat_sys_get_robust_list
diff --git a/arch/mips/kernel/scall64-o32.S b/arch/mips/kernel/scall64-o32.S
index 8dce4c2..e10d5c0 100644
--- a/arch/mips/kernel/scall64-o32.S
+++ b/arch/mips/kernel/scall64-o32.S
@@ -510,7 +510,7 @@ sys_call_table:
 	PTR	sys_splice
 	PTR	sys32_sync_file_range		/* 4305 */
 	PTR	sys_tee
-	PTR	sys_vmsplice
+	PTR	compat_sys_vmsplice
 	PTR	compat_sys_move_pages
 	PTR	compat_sys_set_robust_list
 	PTR	compat_sys_get_robust_list	/* 4310 */
